### PR TITLE
 cedille: init at v1.0.0

### PR DESCRIPTION
**Pull request by mpickering**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by mpickering** [comment]
I should also probably package the elisp code but that can be a separate PR I think? I need to check what agda does.

**Comment by mpickering** [comment]
I packaged the elisp now as well.

Here's an example of how to set up emacs to use this modes.

https://gist.github.com/0971d372954137c4578ae9c06d31414c

```
let
  np = import <nixpkgs> {};
  myEmacsConfig = np.writeText "default.el" ''
;; initialize package
(require 'package)
(package-initialize 'noactivate)
;; load some packages
(setq cedille-path "${np.cedille}/bin")
(require 'cedille-mode)
    '';
    myEmacs = np.emacsWithPackages (epkgs: (with epkgs; [
      (np.runCommand "default.el" {} ''
mkdir -p $out/share/emacs/site-lisp
cp ${myEmacsConfig} $out/share/emacs/site-lisp/default.el
'')
      np.emacsPackages.cedille
      melpaStablePackages.use-package
    ]));
in
  np.mkShell { buildInputs = [ myEmacs ];}
```

File: pkgs/applications/science/logic/cedille/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchFromGitHub, alex, happy, Agda, agdaIowaStdlib,
+  glibcLocales, ghcWithPackages }:
+let
+in
```
Comment (Infinisil) [inline pkgs/applications/science/logic/cedille/default.nix]: empty let in

**Review by infinisil — CHANGES_REQUESTED** [review]
Would be nice to have the emacs mode patched so that it finds the cedille binary by default, see https://github.com/cedille/cedille/blob/0cac62eef32b49af77c475f1b3f4d271f83a0b4b/cedille-mode.el#L32

Also, eval error needs to be fixed, not sure what's up with that

**Comment by mpickering** [comment]
I have no idea why CI fails.

**Comment by mpickering** [comment]
Should be good to merge now.

**Comment by Infinisil** [comment]
Why the hash mismatch on Darwin though?

**Comment by mpickering** [comment]
I don't know but it's a standard use of `fetchFromGitHub`.

**Comment by Infinisil** [comment]
Let's just try again, maybe it'll work now

@GrahamcOfBorg build cedille

**Review by infinisil — APPROVED** [review]
@LnL7 mentioned in #nixos-borg that it might be unicode normalization on macOS, see also https://github.com/NixOS/nixpkgs/pull/48128#issuecomment-429195465

Seems fine to me then :)

**Comment by Infinisil** [comment]
Oh actually, one last thing: The commits are named a bit unconventional. First one should be `agdaIowaStdlib: 18734 -> 1.4.0`, second should be `cedille: init at 1.0.0`, third should be `emacsPackages.cedille: init at 1.0.0`

**Comment by adisbladis** [comment]
@mpickering The `cedille` build is failing both on unstable and 20.03.

**Comment by mpickering** [comment]
@adisbladis Perhaps something which should be fixed by #76653 ? I have not used this package since I packaged it.

**Comment by adisbladis** [comment]
@mpickering Nope, it's not fixed by #76653.

**Pull request merged**